### ISSUE TITLE
Consider replacing StructArrays with TypedTables

### ISSUE THREAD

**Issue by epatters** [AlgebraicJulia/Catlab.jl#330]
We should consider whether it's worth replacing [StructArrays.jl](https://github.com/JuliaArrays/StructArrays.jl) with [TypedTables.jl](https://github.com/JuliaData/TypedTables.jl) in the C-sets implementation.

The packages have similar interfaces, compatible with [Tables.jl](https://github.com/JuliaData/Tables.jl), and similar implementations, based on a named tuple of arrays. Thus it should be very straightforward to replace one with the other. On the other hand, my understanding, based on conversations with @jpfairbanks and Lyndon White, is that TypedTable specifically aims to be a fully typed alternative to [DataFrames.jl](https://github.com/JuliaData/DataFrames.jl), while StructArrays has a different philosophy. So, since we think of C-sets as a collection of interconnected data frames, it may make more sense to use TypedTables.

**Comment by epatters** [comment]
Thoughts on this @jpfairbanks or @olynch?

**Comment by jpfairbanks** [comment]
I'm pro typedtables. I've been using them for interacting with SQLite.jl and it's pretty convenient.

**Comment by epatters** [comment]
OK, good to know. I haven't actually used TypedTables yet, so I have no firsthand experience.

**Comment by olynch** [comment]
It seems that on an implementation level, they are pretty much identical; TypedTables are just slightly less magical because they don't have to deal with arbitrary structs. TypedTables doesn't support empty tuples, unfortunately, but it might actually be easier to fix that than it was for StructArrays, given that the underlying code is simpler. However, given that we are going to provide stuff that's beyond the tables interface, I don't think it really matters whether the underlying type uses TypedTables or StructArrays in terms of Tables.jl interface.

So, to sum up, I'd say that I'm leaning towards viewing this as pointless code churn, but my mind is open to being changed.

**Comment by jpfairbanks** [comment]
Ideally you could have a CSet backed by any julia array of storage, this would be really handy for [CuArrays](https://juliagpu.github.io/CUDA.jl/dev/usage/array/) and [Distributed Arrays] (https://github.com/JuliaParallel/DistributedArrays.jl) so that you could execute CSet algorithms on GPUs or Distributed Memory Clusters.

**Comment by epatters** [comment]
Yes, that would be ideal! And I actually think it should be pretty straightforward, since after the tables are constructed, we only interact with them through the basic Tables.jl interface. So let's give this a shot.
